pkg/autoscaler/autoscaler.go

```diff
@@ -254,295 +255,412 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
+			Eventually(func() (bool, error) {
+				sizeCount := 0
+				for _, machineSet := range transientMachineSets {
+					current, err := framework.GetMachineSet(client, machineSet.GetName())
+					if err != nil {
+						return false, err
+					}
+					if *current.Spec.Replicas == expectedReplicas {
+						sizeCount++
+					}
+				}
+				return sizeCount == len(transientMachineSets), nil
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue())
 
-			return *ms.Spec.Replicas == expectedReplicas
-		}, framework.WaitLong, pollingInterval).Should(BeTrue())
+			By("Waiting for all Machines in MachineSets to enter Running phase")
+			framework.WaitForMachineSet(client, transientMachineSets[0].GetName())
+			framework.WaitForMachineSet(client, transientMachineSets[1].GetName())
 
-		// explicitly delete the ClusterAutoscaler
-		// this is needed due to the autoscaler tests requiring singleton
-		// deployments of the ClusterAutoscaler.
-		caName := clusterAutoscaler.GetName()
-		Expect(deleteObject(caName, cleanupObjects[caName])).Should(Succeed())
-		delete(cleanupObjects, caName)
-		Eventually(func() bool {
-			By(fmt.Sprintf("Waiting for ClusterAutoscaler to delete."))
-			if ca, err := framework.GetClusterAutoscaler(client, caName); ca == nil && err != nil {
-				return true
+			By("Deleting the workload")
+			Expect(deleteObject(workload.Name, cleanupObjects[workload.Name])).Should(Succeed())
+			delete(cleanupObjects, workload.Name)
+
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Waiting for MachineSet %s replicas to scale in", machineSet.GetName()))
 			}
-			return false
-		}, framework.WaitLong, pollingInterval).Should(BeTrue())
+			expectedReplicas = 1
+			Eventually(func() (bool, error) {
+				sizeCount := 0
+				for _, machineSet := range transientMachineSets {
+					current, err := framework.GetMachineSet(client, machineSet.GetName())
+					if err != nil {
+						return false, err
+					}
+					if *current.Spec.Replicas == expectedReplicas {
+						sizeCount++
+					}
+				}
+				return sizeCount == len(transientMachineSets), nil
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue())
+
+			for _, machineSet := range transientMachineSets {
+				machines, err := framework.GetMachinesFromMachineSet(client, machineSet)
+				Expect(err).NotTo(HaveOccurred())
+				for _, machine := range machines {
+					By(fmt.Sprintf("Checking Machine %s for %s annotation", machine.Name, machineDeleteAnnotationKey))
+				}
+			}
+			Eventually(func() (bool, error) {
+				annotationsfound := 0
+				for _, machineSet := range transientMachineSets {
+					machines, err := framework.GetMachinesFromMachineSet(client, machineSet)
+					if err != nil {
+						return false, err
+					}
+					for _, machine := range machines {
+						if machine.ObjectMeta.Annotations == nil {
+							continue
+						}
+						if _, exists := machine.ObjectMeta.Annotations[machineDeleteAnnotationKey]; exists {
+							annotationsfound++
+						}
```

Comment:
Again, not sure `annotationsfound` is needed, we can just return early if we find one that doesn't meet the condition we are looking for 
```suggestion
						if _, exists := machine.ObjectMeta.Annotations[machineDeleteAnnotationKey]; exists {
							return false, nil
						}
```

Reply:
yup, more leftovers from refactor